### ISSUE TITLE
Waiting for API to be available

### ISSUE THREAD

**Issue by rkitindi-kr** [rancher/rancher#44183]
Hi Guys

I think I have spent a lot of time trying to add cluster (RKE2 v.1.26) into RANCHER (Tried both 2.8 and 2.7.9) but I am getting same message (Waiting for API to be available) and I am not able to add cluster into Rancher. 

Can someone help to understand the following:

What does this error mean, what is missing? on cluster or rancher?

What should troubleshooting steps?

Any suggestions will be appreciated.

Thank you.

---

Here is the list of all pods 

NAMESPACE       NAME                                                        READY   STATUS      RESTARTS     AGE
cattle-system   pod/cattle-cluster-agent-6cdd7bcfb4-x6tr5                   1/1     Running     0            41m
kube-system     pod/cilium-78trs                                            1/1     Running     0            9h
kube-system     pod/cilium-j4r88                                            1/1     Running     0            9h
kube-system     pod/cilium-operator-6fc8f77994-kg2h5                        1/1     Running     0            9h
kube-system     pod/cilium-operator-6fc8f77994-l8zjb                        1/1     Running     0            9h
kube-system     pod/cilium-qvcjq                                            1/1     Running     0            9h
kube-system     pod/cloud-controller-manager-ecs1u                          1/1     Running     0            9h
kube-system     pod/cloud-controller-manager-ecs2u                          1/1     Running     0            9h
kube-system     pod/cloud-controller-manager-ecs3u                          1/1     Running     0            9h
kube-system     pod/etcd-ecs1u                                              1/1     Running     0            9h
kube-system     pod/etcd-ecs2u                                              1/1     Running     0            9h
kube-system     pod/etcd-ecs3u                                              1/1     Running     0            9h
kube-system     pod/helm-install-rke2-cilium-kgkkb                          0/1     Completed   0            9h
kube-system     pod/helm-install-rke2-coredns-65gh7                         0/1     Completed   0            9h
kube-system     pod/helm-install-rke2-ingress-nginx-4g69x                   0/1     Completed   0            9h
kube-system     pod/helm-install-rke2-metrics-server-tkt5v                  0/1     Completed   0            9h
kube-system     pod/helm-install-rke2-multus-wpvkz                          0/1     Completed   0            9h
kube-system     pod/helm-install-rke2-snapshot-controller-88ph7             0/1     Completed   1            9h
kube-system     pod/helm-install-rke2-snapshot-controller-crd-w6t28         0/1     Completed   0            9h
kube-system     pod/helm-install-rke2-snapshot-validation-webhook-bsml8     0/1     Completed   0            9h
kube-system     pod/kube-apiserver-ecs1u                                    1/1     Running     0            9h
kube-system     pod/kube-apiserver-ecs2u                                    1/1     Running     0            9h
kube-system     pod/kube-apiserver-ecs3u                                    1/1     Running     0            9h
kube-system     pod/kube-controller-manager-ecs1u                           1/1     Running     0            9h
kube-system     pod/kube-controller-manager-ecs2u                           1/1     Running     0            9h
kube-system     pod/kube-controller-manager-ecs3u                           1/1     Running     0            9h
kube-system     pod/kube-proxy-ecs1u                                        1/1     Running     0            9h
kube-system     pod/kube-proxy-ecs2u                                        1/1     Running     0            9h
kube-system     pod/kube-proxy-ecs3u                                        1/1     Running     0            9h
kube-system     pod/kube-scheduler-ecs1u                                    1/1     Running     0            9h
kube-system     pod/kube-scheduler-ecs2u                                    1/1     Running     0            9h
kube-system     pod/kube-scheduler-ecs3u                                    1/1     Running     0            9h
kube-system     pod/rke2-coredns-rke2-coredns-autoscaler-79f8d49fcc-4j8mj   1/1     Running     0            9h
kube-system     pod/rke2-coredns-rke2-coredns-dd55f557c-2pjxg               1/1     Running     0            9h
kube-system     pod/rke2-coredns-rke2-coredns-dd55f557c-7zrh9               1/1     Running     0            9h
kube-system     pod/rke2-ingress-nginx-controller-68ckc                     1/1     Running     0            9h
kube-system     pod/rke2-ingress-nginx-controller-k6j94                     1/1     Running     0            9h
kube-system     pod/rke2-ingress-nginx-controller-lkj2r                     1/1     Running     0            9h
kube-system     pod/rke2-metrics-server-ffc646896-zrlkp                     1/1     Running     0            9h
kube-system     pod/rke2-multus-ds-ngmch                                    1/1     Running     2 (9h ago)   9h
kube-system     pod/rke2-multus-ds-nn8dw                                    1/1     Running     2 (9h ago)   9h
kube-system     pod/rke2-multus-ds-tmf6c                                    1/1     Running     2 (9h ago)   9h
kube-system     pod/rke2-snapshot-controller-b477fddb-xmqnr                 1/1     Running     0            9h
kube-system     pod/rke2-snapshot-validation-webhook-7b4594d659-gvqc2       1/1     Running     0            9h

---
I see errors on Rancher pod:

2024/01/23 03:45:27 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled
W0123 03:47:03.632421      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 Cluster is deprecated; use cluster.x-k8s.io/v1beta1 Cluster
W0123 03:47:08.468468      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 MachineHealthCheck is deprecated; use cluster.x-k8s.io/v1beta1 MachineHealthCheck
2024/01/23 03:47:12 [INFO] Stopping cluster agent for c-m-npmvpvvm
2024/01/23 03:47:12 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled
2024/01/23 03:49:37 [INFO] Stopping cluster agent for c-m-npmvpvvm
2024/01/23 03:49:37 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled
2024/01/23 03:50:17 [ERROR] Error during subscribe websocket: close sent
W0123 03:50:35.459376      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 MachineSet is deprecated; use cluster.x-k8s.io/v1beta1 MachineSet
2024/01/23 03:51:42 [INFO] Stopping cluster agent for c-m-npmvpvvm
2024/01/23 03:51:42 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled
W0123 03:52:16.199922      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 Machine is deprecated; use cluster.x-k8s.io/v1beta1 Machine
W0123 03:52:26.633860      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 Cluster is deprecated; use cluster.x-k8s.io/v1beta1 Cluster
2024/01/23 03:53:47 [INFO] Stopping cluster agent for c-m-npmvpvvm
2024/01/23 03:53:47 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled
W0123 03:53:56.020743      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 MachineDeployment is deprecated; use cluster.x-k8s.io/v1beta1 MachineDeployment
W0123 03:55:44.461156      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 MachineSet is deprecated; use cluster.x-k8s.io/v1beta1 MachineSet
2024/01/23 03:55:52 [INFO] Stopping cluster agent for c-m-npmvpvvm
2024/01/23 03:55:52 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled
W0123 03:56:27.471437      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 MachineHealthCheck is deprecated; use cluster.x-k8s.io/v1beta1 MachineHealthCheck
2024/01/23 03:57:24 [INFO] Stopping cluster agent for c-m-npmvpvvm
2024/01/23 03:57:24 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled
W0123 03:57:27.201846      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 Machine is deprecated; use cluster.x-k8s.io/v1beta1 Machine
W0123 03:57:39.635823      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 Cluster is deprecated; use cluster.x-k8s.io/v1beta1 Cluster
2024/01/23 04:00:02 [INFO] Stopping cluster agent for c-m-npmvpvvm
2024/01/23 04:00:02 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled
W0123 04:00:15.022460      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 MachineDeployment is deprecated; use cluster.x-k8s.io/v1beta1 MachineDeployment
2024/01/23 04:02:07 [INFO] Stopping cluster agent for c-m-npmvpvvm
2024/01/23 04:02:07 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled
W0123 04:03:13.202932      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 Machine is deprecated; use cluster.x-k8s.io/v1beta1 Machine
W0123 04:03:15.472708      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 MachineHealthCheck is deprecated; use cluster.x-k8s.io/v1beta1 MachineHealthCheck
2024/01/23 04:04:12 [INFO] Stopping cluster agent for c-m-npmvpvvm
2024/01/23 04:04:12 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled
W0123 04:05:41.462299      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 MachineSet is deprecated; use cluster.x-k8s.io/v1beta1 MachineSet
W0123 04:05:53.637413      32 warnings.go:80] cluster.x-k8s.io/v1alpha3 Cluster is deprecated; use cluster.x-k8s.io/v1beta1 Cluster
2024/01/23 04:06:17 [INFO] Stopping cluster agent for c-m-npmvpvvm
2024/01/23 04:06:17 [ERROR] failed to start cluster controllers c-m-npmvpvvm: context canceled

---

**Comment by rkitindi-kr** [comment]
Looks like this happens only when we add PROXY (which I need) to RANCHER DOCKER container, can someone suggest best way for using proxy in docker container?

**Comment by Ghostwritten** [comment]
rancher server :v2.8.1
![WechatIMG210](https://github.com/rancher/rancher/assets/19782612/ba32d9b1-aca6-4d01-90ec-d3cc7caaea6b)

kubectl apply -f cattle-cluster-agent no error:
```
 rancher-agent]# kubectl get pod -n cattle-system
NAME                                   READY   STATUS    RESTARTS   AGE
cattle-cluster-agent-d8d6c7c7c-p86fz   1/1     Running   0          10s
[root@rke-master01 rancher-agent]# kubectl logs  -n cattle-system cattle-cluster-agent-d8d6c7c7c-p86fz
INFO: Environment: CATTLE_ADDRESS=10.42.0.25 CATTLE_CA_CHECKSUM=d818528e6c91a42ed9573c1cbe4b6e3df067d3ebca8b57efccb8e463306e3760 CATTLE_CLUSTER=true CATTLE_CLUSTER_AGENT_PORT=tcp://10.43.131.144:80 CATTLE_CLUSTER_AGENT_PORT_443_TCP=tcp://10.43.131.144:443 CATTLE_CLUSTER_AGENT_PORT_443_TCP_ADDR=10.43.131.144 CATTLE_CLUSTER_AGENT_PORT_443_TCP_PORT=443 CATTLE_CLUSTER_AGENT_PORT_443_TCP_PROTO=tcp CATTLE_CLUSTER_AGENT_PORT_80_TCP=tcp://10.43.131.144:80 CATTLE_CLUSTER_AGENT_PORT_80_TCP_ADDR=10.43.131.144 CATTLE_CLUSTER_AGENT_PORT_80_TCP_PORT=80 CATTLE_CLUSTER_AGENT_PORT_80_TCP_PROTO=tcp CATTLE_CLUSTER_AGENT_SERVICE_HOST=10.43.131.144 CATTLE_CLUSTER_AGENT_SERVICE_PORT=80 CATTLE_CLUSTER_AGENT_SERVICE_PORT_HTTP=80 CATTLE_CLUSTER_AGENT_SERVICE_PORT_HTTPS_INTERNAL=443 CATTLE_CLUSTER_REGISTRY= CATTLE_INGRESS_IP_DOMAIN=sslip.io CATTLE_INSTALL_UUID=26c8f3d4-ad4d-4412-87e6-2f4ecb3ce63c CATTLE_INTERNAL_ADDRESS= CATTLE_IS_RKE=false CATTLE_K8S_MANAGED=true CATTLE_NODE_NAME=cattle-cluster-agent-d8d6c7c7c-p86fz CATTLE_RANCHER_WEBHOOK_VERSION=103.0.1+up0.4.2 CATTLE_SERVER=https://rancher02.demo.com CATTLE_SERVER_VERSION=v2.8.1
INFO: Using resolv.conf: search cattle-system.svc.cluster.local svc.cluster.local cluster.local nameserver 10.43.0.10 options ndots:5
INFO: https://rancher02.demo.com/ping is accessible
INFO: rancher02.demo.com resolves to 192.168.23.80
INFO: Value from https://rancher02.demo.com/v3/settings/cacerts is an x509 certificate
time="2024-01-30T05:33:30Z" level=info msg="Listening on /tmp/log.sock"
time="2024-01-30T05:33:30Z" level=info msg="Rancher agent version v2.8.1 is starting"
time="2024-01-30T05:33:30Z" level=info msg="EnsureSecretForServiceAccount: waiting for secret [cattle-token-zhtks] to be populated with token"
time="2024-01-30T05:33:30Z" level=info msg="Connecting to wss://rancher02.demo.com/v3/connect/register with token starting with krpt9dn9z5hhd7v68z7kkrtdhk7"
time="2024-01-30T05:33:30Z" level=info msg="Connecting to proxy" url="wss://rancher02.demo.com/v3/connect/register"

```
